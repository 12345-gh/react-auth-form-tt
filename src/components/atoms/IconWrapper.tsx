import React from "react";
import styled from "styled-components";

interface IconWrapperProps {
    width?: number;
    height?: number;
    color?: string;
    children: React.ReactNode;
}

const StyledIcon = styled.svg<IconWrapperProps>`
    width: ${({ width }) => width}px;
    height: ${({ height }) => height}px;
    fill: ${({ color }) => color};
`;

export const IconWrapper: React.FC<IconWrapperProps> = ({
    width = 24,
    height = 24,
    color = "#151D51",
    children,
    ...props
}) => {
    return (
        <StyledIcon
            width={width}
            height={height}
            color={color}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
            {...props}
        >
            {children}
        </StyledIcon>
    );
};
