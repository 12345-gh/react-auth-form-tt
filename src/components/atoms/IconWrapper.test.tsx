import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import { IconWrapper } from "./IconWrapper";

describe("IconWrapper component", () => {
    it("matches the snapshot", () => {
        const { asFragment } = render(
            <IconWrapper width={32} height={32} color="#FF5733" data-testid="icon">
                <circle cx="16" cy="16" r="12" />
            </IconWrapper>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders children correctly", () => {
        render(
            <IconWrapper data-testid="icon">
                <circle cx="12" cy="12" r="10" />
            </IconWrapper>,
        );
        expect(screen.getByTestId("icon")).toBeInTheDocument();
    });

    it("applies default props correctly", () => {
        render(
            <IconWrapper data-testid="icon">
                <circle cx="12" cy="12" r="10" />
            </IconWrapper>,
        );
        const icon = screen.getByTestId("icon");

        expect(getComputedStyle(icon).width).toBe("24px");
        expect(getComputedStyle(icon).height).toBe("24px");
        expect(getComputedStyle(icon).fill).toBe("#151D51");
    });

    it("applies custom props correctly", () => {
        render(
            <IconWrapper width={32} height={32} color="#FF5733" data-testid="icon">
                <circle cx="16" cy="16" r="12" />
            </IconWrapper>,
        );
        const icon = screen.getByTestId("icon");

        expect(getComputedStyle(icon).width).toBe("32px");
        expect(getComputedStyle(icon).height).toBe("32px");
        expect(getComputedStyle(icon).fill).toBe("#FF5733");
    });
});
