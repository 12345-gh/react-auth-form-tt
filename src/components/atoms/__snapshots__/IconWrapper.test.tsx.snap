// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconWrapper component > matches the snapshot 1`] = `
<DocumentFragment>
  <svg
    class="sc-braxZu gdPqBF"
    color="#FF5733"
    data-testid="icon"
    height="32"
    viewBox="0 0 24 24"
    width="32"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="16"
      cy="16"
      r="12"
    />
  </svg>
</DocumentFragment>
`;
